engine/CMakeLists.txt

```diff
@@ -33,24 +34,33 @@
 
 CMAKE_MINIMUM_REQUIRED(VERSION 3.11 FATAL_ERROR)
 
-SET(VEGASTRIKE_VERSION_MAJOR "0")
-SET(VEGASTRIKE_VERSION_MINOR "8")
-SET(VEGASTRIKE_VERSION_PATCH "0")
-# Auto-populate with the git hash of the build
-SET (GIT_ROOT_DIR "${CMAKE_SOURCE_DIR}/..")
-MESSAGE("-- CMAKE_SOURCE_DIR = ${CMAKE_SOURCE_DIR}")
-MESSAGE("-- GIT_ROOT_DIR = ${GIT_ROOT_DIR}")
-IF (DEFINED ENV{GITHUB_SHA} AND NOT "$ENV{GITHUB_SHA}" STREQUAL "")
-    SET (VEGASTRIKE_VERSION_TWEAK "$ENV{GITHUB_SHA}")
+IF (DEFINED ENV{TAG_NAME} AND NOT "$ENV{TAG_NAME}" STREQUAL "")
+    STRING(REGEX REPLACE "^v([0-9]+)\\..*" "\\1" VEGASTRIKE_VERSION_MAJOR "$ENV{TAG_NAME}")
+    STRING(REGEX REPLACE "^v[0-9]+\\.([0-9]+).*" "\\1" VEGASTRIKE_VERSION_MINOR "$ENV{TAG_NAME}")
+    STRING(REGEX REPLACE "^v[0-9]+\\.[0-9]+\\.([0-9]+).*" "\\1" VEGASTRIKE_VERSION_PATCH "$ENV{TAG_NAME}")
+    STRING(REGEX REPLACE "^v[0-9]+\\.[0-9]+\\.[0-9]+(.*)" "\\1" VEGASTRIKE_VERSION_TWEAK "$ENV{TAG_NAME}")
+ELSEIF (DEFINED ENV{GITHUB_SHA} AND NOT "$ENV{GITHUB_SHA}" STREQUAL "")
+    SET(VEGASTRIKE_VERSION_MAJOR "0")
+    SET(VEGASTRIKE_VERSION_MINOR "8")
+    SET(VEGASTRIKE_VERSION_PATCH "0")
+    SET(VEGASTRIKE_VERSION_TWEAK "-$ENV{GITHUB_SHA}")
 ELSE ()
+    SET(VEGASTRIKE_VERSION_MAJOR "0")
+    SET(VEGASTRIKE_VERSION_MINOR "8")
+    SET(VEGASTRIKE_VERSION_PATCH "0")
+    SET (GIT_ROOT_DIR "${CMAKE_SOURCE_DIR}/..")
+    MESSAGE("-- CMAKE_SOURCE_DIR = ${CMAKE_SOURCE_DIR}")
+    MESSAGE("-- GIT_ROOT_DIR = ${GIT_ROOT_DIR}")
     EXECUTE_PROCESS(
-        COMMAND git rev-parse --short HEAD
-        WORKING_DIRECTORY ${GIT_ROOT_DIR}
-        OUTPUT_VARIABLE VEGASTRIKE_VERSION_TWEAK
-        OUTPUT_STRIP_TRAILING_WHITESPACE
+            COMMAND git rev-parse --short HEAD
+            WORKING_DIRECTORY ${GIT_ROOT_DIR}
+            OUTPUT_VARIABLE VEGASTRIKE_VERSION_TWEAK
+            OUTPUT_STRIP_TRAILING_WHITESPACE
     )
+    SET(VEGASTRIKE_VERSION_TWEAK "-$VEGASTRIKE_VERSION_TWEAK")
 ENDIF ()
-
+SET(VEGASTRIKE_VERSION_LONG "${VEGASTRIKE_VERSION_MAJOR}.${VEGASTRIKE_VERSION_MINOR}.${VEGASTRIKE_VERSION_PATCH}${VEGASTRIKE_VERSION_TWEAK}")
+SET(VEGASTRIKE_VERSION_SHORT "${VEGASTRIKE_VERSION_MAJOR}.${VEGASTRIKE_VERSION_MINOR}.${VEGASTRIKE_VERSION_PATCH}")
```

Comment:
FYI - I used the `_STR` to denote that the values were string values where the other data could be used as integer values in other parts of the pipeline.

Reply:
OK, changed it to `VEGASTRIKE_VERSION_LONG_STR` and `VEGASTRIKE_VERSION_SHORT_STR`. Does that work?